### PR TITLE
cmake: rename new packages and module introduced in #95 and #96

### PR DESCRIPTION
**Pull request by ryanofsky**
Rename "LibmultiprocessMacros.cmake" module introduced in #95 to "TargetCapnpSources.cmake" to match the name of the `target_capnp_sources` function it contains. Also install it to:

```sh
  <prefix>/lib/cmake/Libmultiprocess/TargetCapnpSources.cmake
```

instead of:

```sh
  <prefix>/lib/cmake/LibmultiprocessGen/LibmultiprocessMacros.cmake
```

Rename the "Libmultiprocess" and "LibmultiprocessGen" packages introduced in #96 to "LibmultiprocessLib" and "LibmultiprocessBin", so package names are consistent with component names "lib" and "bin", and one package name is not a prefix of the other. Also rename intermediate files to match component names.

|                | New                              | Current                                      |
|----------------|----------------------------------|----------------------------------------------|
| Component name | lib                              | lib                                          |
| Package name   | LibmultiprocessLib               | Libmultiprocess                              |
| Config file    | LibmultiprocessLibConfig.cmake   | Libmultiprocess/LibmultiprocessConfig.cmake  |
| Targets file   | Libmultiprocess/LibTargets.cmake | Libmultiprocess/LibmultiprocessTargets.cmake |

|                | New                              | Current                                            |
|----------------|----------------------------------|----------------------------------------------------|
| Component name | bin                              | bin                                                |
| Package name   | LibmultiprocessBin               | LibmultiprocessGen                                 |
| Config file    | LibmultiprocessBinConfig.cmake   | LibmultiprocessGen/LibmultiprocessGenConfig.cmake  |
| Targets file   | Libmultiprocess/BinTargets.cmake | LibmultiprocessGen/LibmultiprocessGenTargets.cmake |

### CODE REVIEW

**Comment by hebasto** [comment]
After the following commands:
```
$ cmake -B build -DCMAKE_INSTALL_PREFIX=/home/hebasto/INSTALL
$ cmake --build build
$ cmake --install build
```
I have the following installation `lib` subdirectory structure:
```
$ tree /home/hebasto/INSTALL
/home/hebasto/INSTALL
├── bin
│   └── mpgen
├── include
│   ├── mp
│   │   ├── proxy.capnp
│   │   ├── proxy.capnp.h
│   │   ├── proxy.h
│   │   ├── proxy-io.h
│   │   ├── proxy-types.h
│   │   └── util.h
│   └── mpgen.mk
└── lib
    ├── cmake
    │   ├── Libmultiprocess
    │   │   ├── BinTargets.cmake
    │   │   ├── BinTargets-noconfig.cmake
    │   │   ├── LibTargets.cmake
    │   │   ├── LibTargets-noconfig.cmake
    │   │   └── TargetCapnpSources.cmake
    │   ├── LibmultiprocessBinConfig.cmake
    │   └── LibmultiprocessLibConfig.cmake
    ├── libmultiprocess.a
    └── pkgconfig
        └── libmultiprocess.pc

7 directories, 17 files
```

A downstream project being provided with `set(CMAKE_FIND_ROOT_PATH /home/hebasto/INSTALL)` will fail to find `LibmultiprocessBin` and `LibmultiprocessLib` packages with the usual `find_package()` command due to its search procedure described in the [docs](https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure).

In short, paths like `<prefix>/lib/cmake/<PackageName>Config.cmake` are not considered at all.
